lib/load_config.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+require 'yaml'
+
+module BCDiceAPI
+  def self.load_config
+    path = File.expand_path('../config/admin.yaml', __dir__)
+    config = if File.exist?(path)
+        yaml = YAML.load_file(path)
+        {
+          name: yaml['admin_name'],
+          address: yaml['admin_address']
+        }
+      else
+        {
+          name: nil,
+          address: nil
+        }
+      end
+
+    config.merge(
+      {
+        name: ENV['BCDiceAPI_ADMIN_NAME'],
+        address: ENV['BCDiceAPI_ADMIN_ADDRESS']
+      }.compact
+    )
```

Comment:
```suggestion
    config = {}
    
    if File.exist?(path)
      yaml = YAML.load_file(path)
      config = {
        name: yaml['name'],
        url: yaml['url'],
        email: yaml['email'],
      }.compact
    end
    
    config[:name] = ENV['BCDICE_API_ADMIN_NAME'] if ENV['BCDICE_API_ADMIN_NAME']
    config[:url] = ENV['BCDICE_API_ADMIN_URL'] if ENV['BCDICE_API_ADMIN_URL']
    config[:email] = ENV['BCDICE_API_ADMIN_EMAIL'] if ENV['BCDICE_API_ADMIN_EMAIL']

    config
```

初期値を明示して、そこから上書きしていくスタイルの方が読みやすそうです。

Reply:
修正案15行目の compact により、設定されていない場合にこのエンドポイントが何も項目を持たないことになってしまいます。
クライアント側の処理が面倒になると思うので API では明示的に null を返す方がより良いと考えますが、いかがでしょうか。

Reply:
suggestを書いた時点では指定しない場合は、項目自体がなくて良いと思ってそう書いていました。nullを返すよりも、設定されていないものは空白文字`""`を返すというのが良さそうなのですが、どうでしょう。

SlackやGitHubのWebAPIを見てみましたが、設定されていないものは空文字で返すというのが一般的なようです。
- https://api.slack.com/methods/users.list
- https://developer.github.com/v3/users/

Reply:
yamlの方で、各要素を`to_s`して入れておけば `nil` でも勝手に空白になるのと、文字列であることを保証できて一石二鳥かもしれません

Reply:
なるほど、調査までして頂きありがとうございます。
では、設定されていなかった場合は空文字列を入れるようにします。